src/libraries/Microsoft.PowerFx.Interpreter/Values/LambdaFormulaValue.cs

```diff
@@ -15,21 +15,33 @@ internal class LambdaFormulaValue : FormulaValue
     {
         private readonly IntermediateNode _tree;
 
+        private readonly EvalVisitor _runner;
+
+        private EvalVisitorContext Context { get; set; }
+
         // Lambdas don't get a special type. 
         // Type is the type the lambda evaluates too. 
-        public LambdaFormulaValue(IRContext irContext, IntermediateNode node)
+        public LambdaFormulaValue(IRContext irContext, IntermediateNode node, EvalVisitor visitor, EvalVisitorContext context)
             : base(irContext)
         {
             _tree = node;
+            _runner = visitor;
+            Context = context;
         }
 
-        public async ValueTask<FormulaValue> EvalAsync(EvalVisitor runner, EvalVisitorContext context)
+        public async ValueTask<FormulaValue> EvalAsync()
         {
-            runner.CheckCancel();
-            var result = await _tree.Accept(runner, context);
+            _runner.CheckCancel();
+            var result = await _tree.Accept(_runner, Context);
             return result;
         }
 
+        public async ValueTask<FormulaValue> EvalInRowScopeAsync(EvalVisitorContext context)
+        {
+            Context = context;
```

Comment:
> Context = context; [](http://example.com/codeflow?start=11&length=19)

We should avoid mutation - that can lead to very subtle bugs.  All the fields here should be readonly. 

Just pass the context through to Accept.